org.adempiere.base/src/org/idempiere/process/VerifyMigration.java

```diff
@@ -0,0 +1,246 @@
+	/* SeqNo */
+	private int m_SeqNo = 0;
+
+	@Override
+	protected void prepare() {
+		for (ProcessInfoParameter para : getParameter()) {
+			String name = para.getParameterName();
+			switch (name) {
+			case "DateTo":
+				p_DateTo = para.getParameterAsTimestamp();
+				break;
+			default:
+				if (log.isLoggable(Level.INFO))
+					log.log(Level.INFO, "Custom Parameter: " + name + "=" + para.getInfo());
+				break;
+			}
+		}
+	}
+
+	@Override
+	protected String doIt() throws Exception {
+		if (log.isLoggable(Level.INFO))
+			log.info("DateTo" + p_DateTo);
+
+		int cnt = 0;
+
+		cnt = cnt + verifyCustomizationsInChangeLog();
+
+		cnt = cnt + verifyCustomViewColumns();
+
+		addLog(getAD_PInstance_ID(), null, null, Msg.getElement(getCtx(), MPInstance.COLUMNNAME_AD_PInstance_ID) + " " + getAD_PInstance_ID(), MPInstance.Table_ID, getAD_PInstance_ID());
+		return "@Inserted@ " + cnt;
+	}
+
+	/**
+	 * @return number of records inserted in AD_VerifyMigration
+	 */
+	private int verifyCustomizationsInChangeLog() {
+		/* For each LAST customization change log record (changes to official records in official tables)
+		 *   Report if there is a difference between the actual value in the database and the value changed by customization
+		 */
+		int cnt = 0;
+		StringBuilder sql = new StringBuilder();
+		sql.append("SELECT * ");
+		sql.append("FROM   AD_ChangeLog cl ");
+		sql.append("WHERE  cl.AD_Client_ID = 0 ");
+		sql.append("       AND cl.AD_Table_ID<=").append(MTable.MAX_OFFICIAL_ID);
+		sql.append("       AND cl.Record_ID<=").append(MTable.MAX_OFFICIAL_ID);
+		sql.append("       AND cl.EventChangeLog = 'U' ");
+		sql.append("       AND cl.AD_ChangeLog_ID = (SELECT MAX(clm.AD_ChangeLog_ID) ");
+		sql.append("                                 FROM   AD_ChangeLog clm ");
+		sql.append("                                 WHERE  cl.AD_Client_ID = clm.AD_Client_ID ");
+		sql.append("                                        AND cl.AD_Table_ID = clm.AD_Table_ID ");
+		sql.append("                                        AND cl.Record_ID = clm.Record_ID ");
+		sql.append("                                        AND cl.AD_Column_ID = clm.AD_Column_ID ");
+		sql.append("                                        AND cl.EventChangeLog = clm.EventChangeLog) ");
+		sql.append("ORDER  BY cl.AD_Table_ID, ");
+		sql.append("          cl.AD_Column_ID, ");
+		sql.append("          cl.Record_ID");
+		try (PreparedStatement pstmt = DB.prepareStatement(sql.toString(), get_TrxName())) {
+			ResultSet rs = pstmt.executeQuery();
+			while (rs.next()) {
+				MChangeLog cl = new MChangeLog(getCtx(), rs, get_TrxName());
+				MTable table = MTable.get(cl.getAD_Table_ID());
+				MColumn column = MColumn.get(cl.getAD_Column_ID());
+				PO po = table.getPO(cl.getRecord_ID(), get_TrxName());
+				Object currentValue = po.get_Value(column.getColumnName());
+				String expectedValue = cl.getNewValue();
+				if (   (currentValue == null && expectedValue != null)
```

Comment:
The comment explaining the if should be closer to the if statement, it is far from the code where it is used and quite difficult to understand at first sight.

Reply:
not sure if you refer to the comment on line 137.

That comment is not about the if - the comment is the value of the message VM_ExpectedValueDiffersFromCurrentValue, it is there to make it clear the variables used in the Msg.getMsg parsing.

Reply:
about the comment on the opening of the method - I did a new commit repositioning it